app/services/report_actions/report_content.rb

```diff
@@ -81,7 +82,8 @@ def generate_my_modules_content(experiment, experiment_element, selected_my_modu
         my_module_element = save_element!({ 'my_module_id' => my_module.id }, :my_module, experiment_element)
 
         @repositories.each do |repository|
-          next unless my_module.live_and_snapshot_repositories_list.pluck(:id).include?(repository.id)
+          repository = assigned_repository_or_snapshot(my_module, repository)
+          next unless repository
```

Comment:
maybe better solved with `.filter_map` (added in ruby 2.7), instead of using `next unless` to skip empty